src/nsidc/metgen/readers/netcdf_reader.py

```diff
@@ -122,17 +123,35 @@ def spatial_values(netcdf, configuration, gsr) -> list[dict]:
     general-use module.
     """
 
+    if gsr == constants.CARTESIAN:
+        return bounding_rectangle_from_attrs(netcdf)
+    elif gsr == constants.GEODETIC:
+        prefer_bounds = getattr(configuration, "prefer_geospatial_bounds", False)
+        if prefer_bounds:
+            return bounding_rectangle_from_geospatial_bounds(netcdf)
+        else:
+            return bounding_rectangle_from_coordinates(netcdf, configuration)
+    else:
+        raise Exception(f"Unsupported granule spatial representation: {gsr}")
+
+
+def bounding_rectangle_from_coordinates(netcdf, configuration) -> list[dict]:
```

Comment:
We're not creating bounding rectangles for these cases, we're returning a series of points that form a closed polygon.